example/lib/src/counters/riverpod_counter/riverpod_counter_screen_controller.dart

```diff
@@ -0,0 +1,32 @@
+import 'package:mocktail/mocktail.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+part 'riverpod_counter_screen_controller.g.dart';
+
+/// Annotating a class by `@riverpod` defines a new shared state for your application,
+/// accessible using the generated [counterProvider].
+/// This class is both responsible for initializing the state (through the [build] method)
+/// and exposing ways to modify it (cf [increment]).
+@riverpod
+class RiverpodCounterScreenController
+    extends _$RiverpodCounterScreenController {
+  /// Classes annotated by `@riverpod` **must** define a [build] function.
+  /// This function is expected to return the initial state of your shared state.
+  /// It is totally acceptable for this function to return a [Future] or [Stream] if you need to.
+  /// You can also freely define parameters on this method.
+  @override
+  int build() => 0;
+
+  void increment() => state++;
+}
+
+// Mock for tests.
+class MockRiverpodCounterScreenController
+    extends _$RiverpodCounterScreenController
+    with Mock
+    implements RiverpodCounterScreenController {
+  final int initialState;
+  MockRiverpodCounterScreenController(this.initialState);
+  @override
+  int build() => initialState;
+}
```

Comment:
Usually we create it in a file with tests

Reply:
Sadly i can't do it because of this https://riverpod.dev/docs/essentials/testing#mocking-notifiers

> For this to work, your mock will have to be placed in the same file as the Notifier you are mocking. Otherwise you would not have access to the _$MyNotifier class.